src/background/stores/tab-store.ts

```diff
@@ -54,9 +54,15 @@ export class TabStore extends BaseStoreImpl<TabStoreData> {
     };
 
     private onNewTabCreated = (payload: Tab): void => {
-        this.state.id = payload.id;
-        this.state.title = payload.title;
-        this.state.url = payload.url;
+        if (payload.id !== null && payload.id !== undefined) { 
```

Comment:
I would lean against using individual conditionals for each of the individual `Tab` properties like this. With the individual conditionals, if a `payload` were to ever contain a mix of null and non-null properties, it would result in `state` being updated into a torn state where some state information is from the new tab and some is from the previous tab. I think that's probably not behavior we want; I'd expect that we probably want `state` to be self-consistent, so I think it probably makes more sense to pick one of:

a) if we get an incomplete new tab notification, assume it's either invalid or irrelevant and that it's safer for us to keep the old state as-is. Log an error and don't mutate the state.
b) if we get an incomplete new tab notification, assume that the old information has been invalidated, whether or not the new information is complete. Put incomplete information in the state and make sure that consumers of the state are able to handle incomplete information.

I think we'd need to research/test whether there are any conditions in which the browser will actually report new tab events with incomplete state, and think about how we want to handle those conditions. If we think there are realistic situations where the browser can report new tab events with incomplete information, we probably need to use approach b. If we are convinced that the browser won't report such events, approach a probably makes more sense.